include/openssl/core_numbers.h

```diff
@@ -313,6 +324,65 @@ OSSL_CORE_MAKE_FUNC(int, OP_kdf_get_ctx_params,
 OSSL_CORE_MAKE_FUNC(int, OP_kdf_set_ctx_params,
                     (void *kctx, const OSSL_PARAM params[]))
 
+/* RAND */
+
+# define OSSL_FUNC_RAND_NEWCTX                        1
+# define OSSL_FUNC_RAND_FREECTX                       2
+# define OSSL_FUNC_RAND_INSTANTIATE                   3
+# define OSSL_FUNC_RAND_UNINSTANTIATE                 4
+# define OSSL_FUNC_RAND_GENERATE                      5
+# define OSSL_FUNC_RAND_RESEED                        6
+# define OSSL_FUNC_RAND_SEED                          7
+# define OSSL_FUNC_RAND_LOCK                          8
+# define OSSL_FUNC_RAND_UNLOCK                        9
+# define OSSL_FUNC_RAND_GETTABLE_PARAMS              10
+# define OSSL_FUNC_RAND_GETTABLE_CTX_PARAMS          11
+# define OSSL_FUNC_RAND_SETTABLE_CTX_PARAMS          12
+# define OSSL_FUNC_RAND_GET_PARAMS                   13
+# define OSSL_FUNC_RAND_GET_CTX_PARAMS               14
+# define OSSL_FUNC_RAND_SET_CTX_PARAMS               15
+
+OSSL_CORE_MAKE_FUNC(void *, OP_rand_newctx,
+                    (void *provctx, const OSSL_RAND_SEED *parent,
+                     void *parent_ctx, CRYPTO_RWLOCK *lock))
+OSSL_CORE_MAKE_FUNC(void, OP_rand_freectx, (void *rctx))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_instantiate,
+                    (void *drbg, const void *pstr, size_t pstr_len))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_uninstantiate, (void *drbg))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_generate,
+                    (void *rctx, void *out, size_t outlen, const void *addin,
+                     size_t addin_len, int prediction_resistance))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_reseed,
+                    (void *rctx, const void *addin, size_t addin_len,
+                     int prediction_resistance))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_seed,
+                    (void *rctx, const void *ent, size_t ent_len,
+                     double randomness))
+OSSL_CORE_MAKE_FUNC(const OSSL_PARAM *, OP_rand_gettable_params, (void))
+OSSL_CORE_MAKE_FUNC(const OSSL_PARAM *, OP_rand_gettable_ctx_params, (void))
+OSSL_CORE_MAKE_FUNC(const OSSL_PARAM *, OP_rand_settable_ctx_params, (void))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_get_params, (OSSL_PARAM params[]))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_get_ctx_params,
+                    (void *rctx, OSSL_PARAM params[]))
+OSSL_CORE_MAKE_FUNC(int, OP_rand_set_ctx_params,
+                    (void *rctx, const OSSL_PARAM params[]))
+
+/*
+ * Typedef to allow random number generators to be chained between providers
+ * without libcrypto being in the middle.  This requires the ability for one
+ * provider to directly access callbacks from another.
+ */
```

Comment:
I don't see this reflected in the code. All actual calls that I can find happen in libcrypto. What am I missing?
If all the calls are *supposed* to happen in libcrypto, then I don't see why this structure is defined here...
(mind you, a structure like this being shared between libcrypto and providers is a big no-no to begin with)

Reply:
The connection happens in libcrypto, the calling will be in the provider.  Haven't got that far yet...

The existence of this structure boils down to the question of being able to fetch OSSL_DISTPACH tables or not.

Reply:
The `get_entropy` call in the DRBG will need to call the DRBG's parent for seed material or something similar.

Reply:
So basically, what's currently happening is that you pass a method structure between libcrypto and provider.  This violates the principle we've had so far, not to share exactly this type of stuff between libcrypto and provider; any change in this structure, or any of the signatures, and that will break compatibility.  We have enough and more flexible shared knowledge through `OSSL_ALGORITHM` / `OSSL_DISPATCH`, do pass that instead and let the provider build up its own method structure.

Reply:
Agreed.

I'm not passing all the methods across and some in this structure won't be required (I added them in the wrong place).